redis/src/main/scala/zio/redis/Input.scala

```diff
@@ -526,114 +573,114 @@ object Input {
       data match {
         case (key, Expire.SetExpireSeconds, duration) =>
-          Chunk(encodeBytes(key), encodeString("EX")) ++ DurationSecondsInput.encode(duration)
+          RespCommand(RespArgument.Key(key), RespArgument.Literal("EX")) ++ DurationSecondsInput.encode(duration)
         case (key, Expire.SetExpireMilliseconds, duration) =>
-          Chunk(encodeBytes(key), encodeString("PX")) ++ DurationMillisecondsInput.encode(duration)
-        case _ => Chunk(encodeBytes(data._1))
+          RespCommand(RespArgument.Key(key), RespArgument.Literal("PX")) ++ DurationMillisecondsInput.encode(duration)
+        case (key, _, _) => RespCommand(RespArgument.Key(key))
       }
   }
 
   final case class GetExAtInput[K: Schema]() extends Input[(K, ExpiredAt, Instant)] {
-    def encode(
-      data: (K, ExpiredAt, Instant)
-    )(implicit codec: BinaryCodec): Chunk[RespValue.BulkString] =
+    def encode(data: (K, ExpiredAt, Instant))(implicit codec: BinaryCodec): RespCommand =
       data match {
         case (key, ExpiredAt.SetExpireAtSeconds, instant) =>
-          Chunk(encodeBytes(key), encodeString("EXAT")) ++ TimeSecondsInput.encode(instant)
+          RespCommand(RespArgument.Key(key), RespArgument.Literal("EXAT")) ++ TimeSecondsInput.encode(instant)
         case (key, ExpiredAt.SetExpireAtMilliseconds, instant) =>
-          Chunk(encodeBytes(key), encodeString("PXAT")) ++ TimeMillisecondsInput.encode(instant)
-        case _ => Chunk(encodeBytes(data._1))
+          RespCommand(RespArgument.Key(key), RespArgument.Literal("PXAT")) ++ TimeMillisecondsInput.encode(instant)
+        case (key, _, _) => RespCommand(RespArgument.Key(key))
       }
   }
 
   case object IdInput extends Input[Long] {
-    def encode(data: Long)(implicit codec: BinaryCodec): Chunk[RespValue.BulkString] =
-      Chunk(encodeString("ID"), encodeString(data.toString))
+    def encode(data: Long)(implicit codec: BinaryCodec): RespCommand =
+      RespCommand(RespArgument.Literal("ID"), RespArgument.Unknown(data.toString))
   }
 
   case object UnblockBehaviorInput extends Input[UnblockBehavior] {
-    def encode(data: UnblockBehavior)(implicit codec: BinaryCodec): Chunk[RespValue.BulkString] =
-      Chunk.single(encodeString(data.stringify))
+    def encode(data: UnblockBehavior)(implicit codec: BinaryCodec): RespCommand =
+      RespCommand(RespArgument.Unknown(data.stringify))
   }
 
   final case class Varargs[-A](input: Input[A]) extends Input[Iterable[A]] {
-    def encode(data: Iterable[A])(implicit codec: BinaryCodec): Chunk[RespValue.BulkString] =
-      data.foldLeft(Chunk.empty: Chunk[RespValue.BulkString])((acc, a) => acc ++ input.encode(a))
+    def encode(data: Iterable[A])(implicit codec: BinaryCodec): RespCommand =
+      data.foldLeft(RespCommand.empty)((acc, a) => acc ++ input.encode(a))
   }
 
   final case class EvalInput[-K, -V](inputK: Input[K], inputV: Input[V]) extends Input[(String, Chunk[K], Chunk[V])] {
```

Comment:
Controversial idea. What Is interface ideally wanted: inputK would have type KeyInput[K] that inherit Input[K]  and we can translate there any dependency, but we cannot to do that cause we have to have contramap that in the inherited interface that return main trait Input. 🤔 
@mijicd What do you think?